tests/admin_views/tests.py

```diff
@@ -5046,11 +5052,13 @@ def test_related_popup_incorrect_close(self):
 
         self.selenium.switch_to.window(test2_window)
         self.selenium.find_element_by_xpath('//input[@value="Save"]').click()
+        self.wait_until(lambda d: len(d.window_handles) == 2, 1)
```

Comment:
Can we use [number_of_windows_to_be()](https://www.selenium.dev/selenium/docs/api/py/webdriver_support/selenium.webdriver.support.expected_conditions.html#selenium.webdriver.support.expected_conditions.number_of_windows_to_be)?
```suggestion
        self.wait_until(ec.number_of_windows_to_be(2))
```
Also, we can probably chop `self.assertEqual(len(self.selenium.window_handles), 2)` in the next line.

Reply:
> Also, we can probably chop self.assertEqual(len(self.selenium.window_handles), 2) in the next line.

The `wait_until` can timeout without the condition being true, so the assertion catches the closing not working. (That's not likely, but it's what the test case is about...)